tools/@aws-cdk/construct-metadata-updater/lib/metadata-updater.ts

```diff
@@ -661,4 +689,78 @@ export class MethodsUpdater extends MetadataUpdater {
     sourceFile.saveSync();
     return updated;
   }
-}
\ No newline at end of file
+}
+
+/**
+ * Class to parse and update the metadata of enum-like classes.
+ * These are classes which are similar to enums, but map to classes rather than
+ * primitive types.
+ */
+export class EnumLikeUpdater extends MetadataUpdater {
+  constructor(dir: string) {
+    super(dir);
+  }
+
+   /**
+   * Parse the repository for any enum-like classes and generate a JSON blueprint.
+   */
+  public execute(): void {
+    const enumlikeBlueprint: Record<string, Record<string, string[]>> = {};
+
+    // Retrieve enum-like classes
+    this.project.getSourceFiles().forEach((sourceFile) => {
+      const sourceFileName: string = sourceFile.getFilePath().split("/aws-cdk/")[1]
+      let fileBlueprint: Record<string, string[]> = {};
+      sourceFile.forEachChild((node) => {
+        if (node instanceof ClassDeclaration) {
+          const className = node.getName();
+          if (className) {
+            node.forEachChild((classField) => {
+              if (classField instanceof PropertyDeclaration) {
+                // enum-likes have `public static readonly` attributes that map to non-string values
+                if (classField.getText().startsWith("public static readonly") && classField.getInitializer()?.getKind() !== SyntaxKind.StringLiteral) {
+                  // This is an enum-like; add to blueprint
+                  const enumlikeName = classField.getName();
+                  if (!fileBlueprint[className]) {
+                    fileBlueprint[className] = [];
+                  }
+                  fileBlueprint[className].push(enumlikeName);
+                }
+              }
+            });
+            if (Object.values(fileBlueprint).length > 0) {
+              enumlikeBlueprint[sourceFileName] = fileBlueprint;
+            }
+          }
+        }
+      });
+    });
+
+    // Generate the file content
+    const outputPath = path.resolve(
+      __dirname,
+      "../../../../packages/aws-cdk-lib/core/lib/analytics-data-source/enums/module-enumlikes.json"
+    );
+    
+    // Write the generated file
+    this.writeFileContent(outputPath, enumlikeBlueprint);
+    console.log(`Metadata file written to: ${outputPath}`);
+  }
+
+
+
+  /**
+   * Write the file content for the enum metadats.
+   */
+  private writeFileContent(outputPath: string, enumlikes: Record<string, Record<string, string[]>> = {}) {
```

Comment:
This function seems to be duplicated 3 times, we can probably extract it out as a method in the base class and re-use that instead of defining 3 times.